packages/math-input/src/components/keypad/__tests__/mobile-keypad.test.tsx

```diff
@@ -0,0 +1,115 @@
+    it("should render keypad when active", () => {
+        // Arrange
+        // Act
+        const {container} = render(
+            <MobileKeypad
+                onAnalyticsEvent={async () => undefined}
+                setKeypadActive={(keypadActive: boolean) => undefined}
+                keypadActive={true}
+            />,
+        );
+
+        // Assert
+        expect(container).toMatchSnapshot();
+    });
+
+    it("should not render the keypad when not active", () => {
+        // Arrange
+        // Act
+        const {container} = render(
+            <MobileKeypad
+                onAnalyticsEvent={async () => undefined}
+                setKeypadActive={(keypadActive: boolean) => undefined}
+                keypadActive={false}
+            />,
+        );
+
+        // Assert
+        expect(container).toMatchSnapshot();
+    });
+
+    it("should render the keypad when going from keypadActive=false to keypadActive=true", () => {
+        // Arrange
+        const {rerender} = render(
+            <MobileKeypad
+                onAnalyticsEvent={async () => undefined}
+                setKeypadActive={(keypadActive: boolean) => undefined}
+                keypadActive={false}
+            />,
+        );
+
+        expect(screen.queryAllByRole("button")).toHaveLength(0);
+
+        // Act
+        rerender(
+            <MobileKeypad
+                onAnalyticsEvent={async () => undefined}
+                setKeypadActive={(keypadActive: boolean) => undefined}
+                keypadActive={true}
+            />,
+        );
+
+        // Assert
+        expect(screen.queryAllByRole("tab")).not.toHaveLength(0);
+    });
+
+    it("should fire an 'opened' event when activated", () => {
+        // Arrange
+        const onAnalyticsEvent = jest.fn();
+
+        // Act
+        render(
+            <MobileKeypad
+                onAnalyticsEvent={onAnalyticsEvent}
+                setKeypadActive={(keypadActive: boolean) => undefined}
+                keypadActive={true}
+            />,
+        );
+
+        // Assert
+        expect(onAnalyticsEvent).toHaveBeenCalledWith({
+            type: "math-input:keypad-opened",
+            payload: {
+                virtualKeypadVersion: "MATH_INPUT_KEYPAD_V2",
+            },
+        });
+    });
+
+    it("should fire an 'closed' event when dismissed", async () => {
+        const onAnalyticsEvent = jest.fn();
+
+        // Arrange
+        const {rerender, unmount} = render(
+            <MobileKeypad
+                onAnalyticsEvent={onAnalyticsEvent}
+                setKeypadActive={(keypadActive: boolean) => undefined}
+                keypadActive={true}
+            />,
+        );
+
+        // Act
+        rerender(
+            <MobileKeypad
+                onAnalyticsEvent={onAnalyticsEvent}
+                setKeypadActive={(keypadActive: boolean) => undefined}
+                keypadActive={false}
+            />,
+        );
+        unmount();
```

Comment:
I am a bit unsure of this. This was the only way I could get the final analytics event to fire and "flush" before the end of the test.

Reply:
Seems right to me. :)